enterprise/cmd/repo-updater/internal/authz/perms_syncer.go

```diff
@@ -301,18 +301,23 @@ func (s *PermsSyncer) getUserGitHubAppInstallations(ctx context.Context, acct *e
 	return installations, nil
 }
 
+type fetchUserPermsViaExternalAccountsResults struct {
+	repoIDs      []uint32
+	subRepoPerms map[api.ExternalRepoSpec]*authz.SubRepoPermissions
+}
```

Comment:
We also sync permissions based on repositories, see method [`syncRepoPerms`](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/enterprise/cmd/repo-updater/internal/authz/perms_syncer.go?L715)

Should we extend this type to include that as well? E.g. something like:
```suggestion
type PermsResults struct {
	repoIDs      []uint32
	userIDs      []uint32
	subRepoPerms map[api.ExternalRepoSpec]*authz.SubRepoPermissions
}
```

Reply:
User and repo perms also return very different things, i.e. in user-centric sync we _only_ return repo lists and in repo-centric sync we _only_ return user lists, creating a struct to carry both would just be a mutually exclusive result set